commit 2dde9824c4815d5281d387409510cecd72a0a2d4 (refs/changes/68/568/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-10-01T01:44:48+00:00 (3 years ago)
    
    joyent/node-cueball#32 reshuffle on empty pool results in crash
    Reviewed by: Joshua M. Clulow <jmc@joyent.com>

diff --git a/lib/pool.js b/lib/pool.js
index 187b1ce..98a97ab 100644
--- a/lib/pool.js
+++ b/lib/pool.js
@@ -334,6 +334,8 @@ CueBallConnectionPool.prototype.isDeclaredDead = function (backend) {
 };
 
 CueBallConnectionPool.prototype.reshuffle = function () {
+	if (this.p_keys.length <= 1)
+		return;
 	var taken = this.p_keys.pop();
 	var idx = Math.floor(Math.random() * (this.p_keys.length + 1));
 	this.p_keys.splice(idx, 0, taken);
diff --git a/package.json b/package.json
index 8a9ca08..3fc9fe4 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "1.0.0",
+  "version": "1.0.1",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
